esmvalcore/preprocessor/_derive/intppgt.py

```diff
@@ -0,0 +1,89 @@
+"""Derivation of variable `intppgt`."""
+import iris
+import numpy as np
+
+from ._baseclass import DerivedVariableBase
+
+
+def calculate_total_flux(intpp_cube, cube_area):
+    """
+    Calculate the area of unmasked cube cells.
+
+    Requires a cube with two spacial dimensions. (no depth coordinate).
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        Data Cube
+    cube_area: iris.cube.Cube
+        Cell area Cube
+
+    Returns
+    -------
+    numpy.array:
+        An numpy array containing the total flux of CO2.
+    """
+    data = []
+    times = intpp_cube.coord('time')
+
+    intpp_cube.data = np.ma.array(intpp_cube.data)
+    for time_itr in np.arange(len(times.points)):
```

Comment:
Would it be possible to use `iris.analysis.SUM` to aggregate over the required dimensions?